java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -2552,6 +2606,27 @@ private static ColumnVector from(DType type, Object dataArray) {
         case FLOAT64:
           ret = ColumnVector.fromBoxedDoubles((Double[]) dataArray);
           break;
+        case DECIMAL32:
+        case DECIMAL64:
+          int scale = type.getScale();
+          if (dataArray instanceof Integer[]) {
+            BigDecimal[] data = Arrays.stream(((Integer[]) dataArray))
+                .map((i) -> i == null ? null : BigDecimal.valueOf(i, -scale))
```

Comment:
With unscaled values provided, why use `BigDecimal` instead of using the much cheaper unscaled value decimal constructors?

Reply:
Because unscaled value constructors can not support null values, since it is for primitive values.

Reply:
> unscaled value constructors can not support null values

That seems like a problem with the value constructors that needs a followup.  We support value constructors for Integer values that can accept null values, so why can't we do the same for unscaled decimal values?  Having to go through `BigDecimal` to workaround a lack of functionality in the value constructor for decimals is not ideal.

Reply:
@jlowe Yes. I created a new issue #6795 to follow this flaw.